### PR TITLE
Added fix for Mach2 VTX regarding Issue #27

### LINKED ISSUE #27: vtx_akk_hack doesn't seem to support RDQ Mach 2
Noticed that whatever I tried I couldn't get the legacy hack to work with my racedayquads mach 2 vtx using a known good configuration (on my omnibus pro corner f4).  I was also using softserial to get it working but this wasn't a problem before.

Reproducible by attempting to make the akk_hack setting work with any (i'd guess) mach 2 vtx and attempting smart audio control via osd.  *sometimes* you can actually see the menu to choose channel / power level but setting those values doesn't actually change the vtx.

This was on the 3.4.1 release of butterflight.

I know this bug is not hardware related because I forked butterflight, checked it out, ran
`git revert -m 1 30f3152`
(this is the merge commit for the betaflight pr that breaks mach 2 compatability)
built this and flashed it to my pro corner and it worked like a charm even over soft serial.  I of course ran into some conflicts but I always chose the incoming changes in those 3 cases and had no trouble building.  Of course I may have broken some stuff like setting power level on disarm and setting custom frequency but I think it's evidence enough this feature may require some more attention.

here's my diff:
```# diff

# version
# ButterFlight / OMNIBUSF4SD (OBSD) 3.4.1 Mar  4 2018 / 02:00:33 (1cd8f73) MSP API: 1.38

# name

# resources
resource SERIAL_TX 1 NONE
resource SERIAL_TX 11 A09

# mixer

# feature
feature SOFTSERIAL

# beeper

# map
map TAER1234

# serial
serial 2 64 115200 57600 0 115200
serial 30 2048 115200 57600 0 115200

# led

# color

# mode_color

# aux

# adjrange

# rxrange

# vtx

# rxfail

# master
set rssi_channel = 6
set serialrx_provider = SBUS
set osd_vtx_channel_pos = 2282

# profile
profile 0


# rateprofile
rateprofile 0


#
 ```

### PR DESCRIPTION
**Pull request by kidBrazil**
Added conditions to change SA_QSIZE and SERIAL_STOPBITS based on the AKK fix we last implemented.

Hopefully it fixes #27 and restores SA functionality. 

We will need testers as I don't have SA enabled VTX

### CODE REVIEW

File: src/main/io/vtx_smartaudio.c [hunk 1]
```diff
@@ -497,8 +497,11 @@ typedef struct saCmdQueue_s {
     uint8_t *buf;
     int len;
 } saCmdQueue_t;
-
-#define SA_QSIZE 6     // 1 heartbeat (GetSettings) + 2 commands + 1 slack
+if (vtxSettingsConfig()->akkStyleEndFrame) {
+  #define SA_QSIZE 4
```
Comment (orneryd) [inline src/main/io/vtx_smartaudio.c]: can't have defines in runtime

**Comment by kidBrazil** [comment]
Thanks man,  I will get it fixed up!

**Comment by kidBrazil** [comment]
Will look at it in the morning, too tired to keep my eyes open

**Comment by apocolipse** [comment]
@kidBrazil Travis failing

**Comment by kidBrazil** [comment]
Polled Butter slack - cant get any confirmation that this is actually an issue -When polled the slack responded that those with old SA were able to get it to work just fine with the Hack so there should be no need to change SA_QSIZE

**Comment by TheAdamizer** [comment]
All I saw was someone with an AKK confirm that the hack worked for them.  This is still very reproducible for me and still very much an issue.

**Pull request closed without merging**